src/ai/backend/client/cli/session/lifecycle.py

```diff
@@ -791,31 +794,33 @@ def status_history(session_id):
         kernel = session.ComputeSession(session_id)
         try:
             status_history = kernel.get_status_history().get("result")
-            print_info(f"status_history: {status_history}")
-            if (preparing := status_history.get("preparing")) is None:
-                result = {
-                    "result": {
-                        "seconds": 0,
-                        "microseconds": 0,
-                    },
-                }
-            elif (terminated := status_history.get("terminated")) is None:
-                alloc_time_until_now: timedelta = datetime.now(tzutc()) - isoparse(preparing)
-                result = {
-                    "result": {
-                        "seconds": alloc_time_until_now.seconds,
-                        "microseconds": alloc_time_until_now.microseconds,
-                    },
-                }
+
+            prev_time = None
+
+            for status_record in status_history:
+                timestamp = datetime.fromisoformat(status_record["timestamp"])
+
+                if prev_time:
+                    time_diff = timestamp - prev_time
+                    status_record["time_elapsed"] = str(time_diff)
+
+                prev_time = timestamp
+
+            ctx.output.print_list(
+                status_history,
+                [FieldSpec("status"), FieldSpec("timestamp"), FieldSpec("time_elapsed")],
+            )
+
+            if (preparing := get_first_timestamp_for_status(status_history, "PREPARING")) is None:
+                elapsed = timedelta()
+            elif (
+                terminated := get_first_timestamp_for_status(status_history, "TERMINATED")
+            ) is None:
+                elapsed = datetime.now(tzutc()) - preparing
             else:
-                alloc_time: timedelta = isoparse(terminated) - isoparse(preparing)
-                result = {
-                    "result": {
-                        "seconds": alloc_time.seconds,
-                        "microseconds": alloc_time.microseconds,
-                    },
-                }
-            print_done(f"Actual Resource Allocation Time: {result}")
+                elapsed = terminated - preparing
+
+            print_done(f"Actual Resource Allocation Time: {elapsed.total_seconds()}")
```

Comment:
You've probably kept the existing behavior based on `dict`, but since you've changed the structure based on `list`, shouldn't you show the whole history change?